command/plugin.go

```diff
@@ -0,0 +1,168 @@
+	"github.com/mitchellh/packer/provisioner/windows-restart"
+	windowsshell "github.com/mitchellh/packer/provisioner/windows-shell"
+)
+
+type PluginCommand struct {
+	Meta
+}
+
+var Builders = map[string]packer.Builder{
+	"amazon-chroot":   new(chroot.Builder),
+	"amazon-ebs":      new(ebs.Builder),
+	"amazon-instance": new(instance.Builder),
+	"digitalocean":    new(digitalocean.Builder),
+	"docker":          new(docker.Builder),
+	"file":            new(filebuilder.Builder),
+	"googlecompute":   new(googlecompute.Builder),
+	"null":            new(null.Builder),
+	"openstack":       new(openstack.Builder),
+	"parallels-iso":   new(parallelsiso.Builder),
+	"parallels-pvm":   new(parallelspvm.Builder),
+	"qemu":            new(qemu.Builder),
+	"virtualbox-iso":  new(virtualboxiso.Builder),
+	"virtualbox-ovf":  new(virtualboxovf.Builder),
+	"vmware-iso":      new(vmwareiso.Builder),
+	"vmware-vmx":      new(vmwarevmx.Builder),
+}
+
+var Provisioners = map[string]packer.Provisioner{
+	"ansible-local":     new(ansiblelocal.Provisioner),
+	"chef-client":       new(chefclient.Provisioner),
+	"chef-solo":         new(chefsolo.Provisioner),
+	"file":              new(fileprovisioner.Provisioner),
+	"powershell":        new(powershell.Provisioner),
+	"puppet-masterless": new(puppetmasterless.Provisioner),
+	"puppet-server":     new(puppetserver.Provisioner),
+	"salt-masterless":   new(saltmasterless.Provisioner),
+	"shell":             new(shell.Provisioner),
+	"shell-local":       new(shelllocal.Provisioner),
+	"windows-restart":   new(restart.Provisioner),
+	"windows-shell":     new(windowsshell.Provisioner),
+}
+
+var PostProcessors = map[string]packer.PostProcessor{
+	"artifice":      new(artifice.PostProcessor),
+	"atlas":         new(atlas.PostProcessor),
+	"compress":      new(compress.PostProcessor),
+	"docker-import": new(dockerimport.PostProcessor),
+	"docker-push":   new(dockerpush.PostProcessor),
+	"docker-save":   new(dockersave.PostProcessor),
+	"docker-tag":    new(dockertag.PostProcessor),
+	"vagrant":       new(vagrant.PostProcessor),
+	"vagrant-cloud": new(vagrantcloud.PostProcessor),
+	"vsphere":       new(vsphere.PostProcessor),
+}
+
+func (c *PluginCommand) Run(args []string) int {
+	// This is an internal call so we're not going to do much error checking.
+	// If there's a problem we'll usually just crash.
+	log.Printf("args: %#v", args)
+	if len(args) != 1 {
+		c.Ui.Error("Wrong number of args")
+		os.Exit(1)
+	}
+
+	// Plugin should be called like "packer-builder-amazon-ebs" so we'll take it
+	// apart.
+	parts := strings.Split(args[0], "-")
+	pluginType := parts[1]
+	pluginName := ""
+	// Post-processor is split so we'll so some magic here. We could use a
+	// regexp but this is simpler.
+	if pluginType == "post" {
+		pluginType = strings.Join(parts[1:2], "-")
+		pluginName = strings.Join(parts[3:], "-")
+	} else {
+		pluginName = strings.Join(parts[2:], "-")
+	}
+
+	server, err := plugin.Server()
+	if err != nil {
+		panic(err)
```

Comment:
I can add some more friendly error messaging but users can't call the plugin on their own. Currently if a user tries to run a plugin packer crashes and says "don't do that".